python/django/security/audit/avoid-insecure-deserialization.yaml

```diff
@@ -14,38 +14,43 @@ rules:
     languages:
       - python
     severity: ERROR
-    patterns:
-      - pattern-inside: |
-          def $X(..., request, ...):
-            ...
-      - pattern-either:
-          - pattern: |
-              $VAR = <... request.$Y.get(...) ...>
+    mode: taint
+    pattern-sources:
+    - pattern-either:
+      - patterns:
+        - pattern-inside: |
+            def $INSIDE(..., $PARAM, ...):
               ...
-              pickle.$FUNC(<... $VAR ...>)
-          - pattern: pickle.$FUNC(<... request.$Y.get(...) ...>)
+        - pattern: |
```

Comment:
For Django, user data almost always comes from the `request` object, so we can limit scope to functions with `request` as a parameter rather than any arbitrary paramter

Reply:
ah good catch, I'll also change the pattern to a request.$PATH.get(...) call

Reply:
I'm not 100% certain of the precise answer, but I've always seen in Django a parameter called `request` used to indicate the incoming request. It's probably not a big deal, but it using a `pattern-inside` clause to scope to a parameter named `request` might help with stray, random variables that also happened to be named `request`. 🤷‍♂️  I'll  let you make the final decision so I'll go ahead and approve :)